core/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -303,6 +306,27 @@ public int getBatchedReduceSize() {
     }
 
     /**
+     * Returns the number of shard requests that should be executed concurrently. This value should be used as a protection mechanism to
+     * reduce the number of shard reqeusts fired per high level search request. Searches that hit the entire cluster can be throttled
+     * with this number to reduce the cluster load. The default is <tt>256</tt>
+     */
+    public int getMaxNumConcurrentShardRequests() {
+        return maxNunConcurrentShardRequests;
+    }
+
+    /**
+     * Sets the number of shard requests that should be executed concurrently. This value should be used as a protection mechanism to
+     * reduce the number of shard reqeusts fired per high level search request. Searches that hit the entire cluster can be throttled
+     * with this number to reduce the cluster load. The default is <tt>256</tt>
+     */
+    public void setMaxNumConcurrentShardRequests(int maxNunConcurrentShardRequests) {
```

Comment:
or maybe rather remove `Num` from the variable name since the same variable seems to be called `maxConcurrentShardRequests` elsewhere and in the dsl, so it would be more consistent?